src/Fixer/ConfigurationDefinitionFixerInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer;
+
+use PhpCsFixer\OptionsResolver;
+
+/**
+ * @deprecated Will be merged into ConfigurableFixerInterface in 3.0
+ */
+interface ConfigurationDefinitionFixerInterface extends ConfigurableFixerInterface
```

Comment:
@juliendufresne 

Please, don't go that far to think about v4, it won't be there for months, if not years.

current state:
We have 2 interfaces for same thing, user will have no idea which one he should use.
We should deprecate the one that we don't want to be used, here, it would be the original one.

Please:
- deprecate `ConfigurableFixerInterface`, as we don't want user to use it

What we will do in v4:
- drop deprecations, so we will remove `ConfigurableFixerInterface`
- we could make BC breakers, like renaming `ConfigurationDefinitionFixerInterface` (maybe into `ConfigurableFixerInterface`)